apps/ledger-live-desktop/src/main/setup.ts

```diff
@@ -7,17 +7,44 @@ import fs from "fs/promises";
 import updater from "./updater";
 import resolveUserDataDirectory from "~/helpers/resolveUserDataDirectory";
 import path from "path";
+import { InMemoryLogger } from "./logger";
+
 ipcMain.on("mainCrashTest", () => {
   captureException(new Error("CrashTestMain"));
 });
+
 ipcMain.on("updater", (e, type) => {
   updater(type);
 });
+
+/**
+ * Saves logs from the renderer thread and logs recorded from the internal thread to a file.
+ */
 ipcMain.handle(
   "save-logs",
-  async (event, path: Electron.SaveDialogReturnValue, experimentalLogs: string) =>
-    !path.canceled && path.filePath && fs.writeFile(path.filePath, experimentalLogs),
+  async (_event, path: Electron.SaveDialogReturnValue, rendererLogs: unknown[]) => {
+    if (!path.canceled && path.filePath) {
+      const inMemoryLogger = InMemoryLogger.getLogger();
+      const internalLogs = inMemoryLogger.getLogs();
+      console.log(
+        `Saving ${rendererLogs.length} logs from the renderer thread and ${internalLogs.length} logs from the internal thread`,
+      );
+
+      // Merging according to a `date` (internal logs) / `timestamp` (most of renderer logs) does not seem necessary.
+      // Simply pushes all the internal logs after the renderer logs.
+      // Note: this is not respecting the `EXPORT_MAX_LOGS` env var, but this is fine.
+      rendererLogs.push(
+        { type: "logs-separator", message: "Logs coming from the internal thread" },
```

Comment:
can you double check that such log file with only "type" and "message" is going to not break the logsviewer?

Reply:
Right ! I'm going to test this

Reply:
<img width="1180" alt="Screenshot 2023-09-18 at 15 44 46" src="https://github.com/LedgerHQ/ledger-live/assets/15096913/e8b34f93-847f-4c94-9778-f96c62bd03c2">

It's still working with it ;)